mk/k3d.mk

```diff
@@ -1,28 +1,30 @@
-CI_K3D_VERSION ?= v4.4.5
+CI_K3D_VERSION ?= v5.2.0
 
 K3D_PATH := $(CI_TOOLS_DIR)/k3d
 
 KUMA_MODE ?= standalone
 KUMA_NAMESPACE ?= kuma-system
 PORT_PREFIX := $$(( $(subst kuma-,30,$(KIND_CLUSTER_NAME)) - 1 ))
 
+.PHONY: k3d/network/create
+k3d/network/create:
+	@touch $(BUILD_DIR)/k3d_network.lock
+	@flock -x $(BUILD_DIR)/k3d_network.lock -c 'docker network create -d=bridge -o com.docker.network.bridge.enable_ip_masquerade=true --ipv6 --subnet "fd00:fd12:3456::0/64" kind || true;' \
+		|| docker network create -d=bridge -o com.docker.network.bridge.enable_ip_masquerade=true --ipv6 --subnet "fd00:fd12:3456::0/64" kind || true
+	@rm -f $(BUILD_DIR)/k3d_network.lock
+
 .PHONY: k3d/start
-k3d/start: ${KIND_KUBECONFIG_DIR}
-	@docker network create \
-		-d=bridge -o com.docker.network.bridge.enable_ip_masquerade=true \
-		--ipv6 --subnet "fd00:fd12:3456::0/64" kind || true
-	@k3d cluster list | grep $(KIND_CLUSTER_NAME) >/dev/null 2>&1 && echo "K3d cluster already running." && exit 0 || \
-		(KUBECONFIG=$(KIND_KUBECONFIG)  \
-		  k3d cluster create "$(KIND_CLUSTER_NAME)" \
-		  	--k3s-server-arg '--no-deploy=traefik' \
-		  	--k3s-server-arg '--disable=metrics-server' \
-		  	--no-lb --no-hostip \
-		  	--network kind \
-		  	--port "$(PORT_PREFIX)80-$(PORT_PREFIX)89:30080-30089" \
-		  	--timeout 120s && \
-		until \
-			KUBECONFIG=$(KIND_KUBECONFIG) kubectl wait -n kube-system --timeout=5s --for condition=Ready --all pods ; \
-		do echo "Waiting for the cluster to come up" && sleep 1; done )
+k3d/start: ${KIND_KUBECONFIG_DIR} k3d/network/create
+	@KUBECONFIG=$(KIND_KUBECONFIG)  \
+		 k3d cluster create "$(KIND_CLUSTER_NAME)" \
```

Comment:
Should you use `$(K3D_PATH)` consistently? Or, if depending on the tools dir being in the path (we do that all the time) seems OK here, remove `${K3D_PATH)`  variable.

Reply:
I think either way is good and we're not really coherent anywhere in all the makefiles so going to keep the simplest syntax (using k3d directly and getting rid of K3D_PATH)